python/sglang/srt/mem_cache/allocator.py

```diff
@@ -508,14 +508,22 @@ def alloc_decode(
                 (last_loc + 2) % self.page_size == seq_lens % self.page_size
             )
 
-        estimated_num_new_pages = (
-            (
-                (seq_lens + self.page_size - 1) // self.page_size
-                - (seq_lens - 1 + self.page_size - 1) // self.page_size
+        # TODO remove the gpu branch if the cpu branch is always faster
+        if seq_lens.shape[0] <= 16:
+            estimated_num_new_pages = sum(
+                (seq_len + self.page_size - 1) // self.page_size
+                - (seq_len - 1 + self.page_size - 1) // self.page_size
+                for seq_len in seq_lens.cpu().tolist()
+            )
+        else:
+            estimated_num_new_pages = (
```

Comment:
IMO we can try
```
seq_lens_cpu = seq_lens.cpu()
estimated_num_new_pages = (
                (
                    (seq_lens_cpu + self.page_size - 1) // self.page_size
                    - (seq_lens_cpu - 1 + self.page_size - 1) // self.page_size
                )
                .sum()
                .item()
            )
```

Reply:
what I worry about is the torch overhead. In bs1 case, seq_lens is only len-1 list, thus calling a dozen torch ops, which go through the slow torch dispatch and python-to-c++ bridges, may be the cause of slow...